haystack/components/generators/openai.py

```diff
@@ -286,11 +284,14 @@ def _check_finish_reason(self, message: ChatMessage) -> None:
         """
         if message.meta["finish_reason"] == "length":
             logger.warning(
-                "The completion for index %s has been truncated before reaching a natural stopping point. "
+                "The completion for index {index} has been truncated before reaching a natural stopping point. "
                 "Increase the max_tokens parameter to allow for longer completions.",
-                message.meta["index"],
+                index=message.meta["index"],
+                finish_reason=message.meta["finish_reason"],
```

Comment:
I can't see the `{finish_reason}` placeholder, is that missing?

Reply:
great catch - this is actually an extra key-value which I found useful to be logged (so it's not interpolated in the message, just attached as key value)